pkg/metrics/metrics.jsx

```diff
@@ -617,7 +617,21 @@ const MetricsHour = ({ startTime, data }) => {
             </dl>);
     }
 
-    for (let minute = 59; minute >= 0; --minute) {
+    let minutes = 60;
+    if (squash) {
+        // When removing of empty minutes is allowed, find the highest valid minute and round to the next 5 minutes
+        let m = 59;
+        for (; m >= 0; --m) {
+            const dataOffset = m * SAMPLES_PER_MIN;
+            const dataSlice = normData.slice(dataOffset, dataOffset + SAMPLES_PER_MIN);
+            if (dataSlice.find(i => i !== null)) {
```

Comment:
This is exactly the same iteratinon/test as the code below already does. But as we have to "rewind" to the previous full 5 minutes, I don't think we can fully avoid it. However, you could do that loop in 5 minute steps instead of 1 minute steps, avoid the arithmetic, and directly work on the `minutes` variable.